src/Renderer.js

```diff
@@ -55,6 +55,73 @@ export default class Renderer {
     return nodeEl;
   }
 
+  // extract all the literals, create clones, and absolutely position 
+  // them at their original locations
+  animateTransition(ast, toBlocks) {
+    var that = this, rootNodes = ast.rootNodes;
+    // take note of the parent elt, CM offsets, and literals from the AST
+    var cm = this.cm, parent = this.cm.getScrollerElement();
+    let {left: offsetLeft, top: offsetTop} = parent.getBoundingClientRect();
+
+    // toDom : AST Node -> DOM Node
+    // given a literal AST node, make a DOM node with the same srcLoc info
+    function toDom(literal) {
+      let el = document.createElement("span");
+      el.appendChild(document.createTextNode(literal.value.toString()));
+      return el;
+    }
+
+    // given a literal, a clone, and whether we're coming from text...
+    // position the clone over the currently-rendered literal
+    function assignClonePosition(literal, clone, fromText) {
+      if(fromText){ // if we're coming from text, fake a literal to get coords
+        literal.el = toDom(literal);
+        var tm = cm.markText(literal.from, literal.to, { replacedWith: literal.el });
+      }
+      // assign the location and other style info
+      let {left, top, width, height} = literal.el.getBoundingClientRect(); 
+      clone.style.top    = (top - offsetTop) + parent.scrollTop  + "px";
+      clone.style.left   = (left- offsetLeft)+ parent.scrollLeft + "px";
+      clone.style.width      = width  + "px";
+      clone.style.height     = height + "px";
+      clone.style.display    = "inline-block";
+      clone.style.position   = "absolute";
+      clone.style.animation  = "none";
+      clone.className        = "transition";
+      if(fromText) { tm.clear(); } // clean up the faked marker
+    }
+
+    // 1) get all the literals from the AST, and make clones of them
+    let literals = rootNodes.reduce((acc, r) => {
+      return acc.concat(Array.from(r).filter((n) => n.type==="literal"));
+    }, []);
+    let clones = literals.map(toDom);
+
+    // 2) move each clone to the *origin* location of the corresponding literal 
+    literals.forEach(function(literal, i) {
+      assignClonePosition(literal, clones[i], toBlocks);
+      parent.appendChild(clones[i]);
+    });
+
+    // 3) render or clear the original AST
+    if(toBlocks) { 
+      rootNodes.forEach(r => { that.render(r);  r.el.style.animationName = "fadein"; });
+    } else { 
+      cm.getAllMarks().forEach(marker => marker.clear()); 
+    }
+
+    // 4) move each clone to the *destination* location of the corresponding literal 
+    literals.forEach((literal, i) => {
+      assignClonePosition(literal, clones[i], !toBlocks);
+    });
+
+    // 5) clean up after ourselves
+    setTimeout(function() {
+      for (let node of rootNodes) { if(node.el) node.el.style.animationName = ""; }
+      for (let c of clones) { if(c) c.remove(); }
+    }, 1000);
```

Comment:
you might want to note that this 1000 value should match up with what is in the css file.